paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/cron/PaytrailStatusCheckController.java

```diff
@@ -0,0 +1,229 @@
+
+    @Autowired
+    PaymentPaytrailService paymentPaytrailService;
+
+    @Autowired
+    private PaytrailPaymentReturnValidator paytrailPaymentReturnValidator;
+
+    @Autowired
+    private SaveHistoryService saveHistoryService;
+
+    @Autowired
+    private SendNotificationService sendNotificationService;
+
+    @Autowired
+    private CommonServiceConfigurationClient commonServiceConfigurationClient;
+
+    @Value("${productmapping.service.url:http://product-mapping-api:8080/productmapping/}")
+    private String productMappingServiceUrl;
+
+
+    @GetMapping(value = "/synchronize-paytrail-payment-status", produces = MediaType.TEXT_PLAIN_VALUE)
+    public ResponseEntity<String> checkAndSynchronizePaytrailPaymentStatus(
+            @Parameter( description = "Datetime after which to check the payments. Defaults to one day ago. Example: 2025-05-10T00:00:00" )
+            @RequestParam(value = "createdAfter", required = false) String createdAfter,
+            @Parameter( description = "Datetime before which to check the payments. Defaults to now - 5 minutes. Example 2025-05-12T12:00:00" )
+            @RequestParam(value = "createdBefore", required = false) String createdBefore
+    ) {
+        boolean sendErrorEmail = false;
+        ObjectMapper objectMapper = new ObjectMapper();
+        List<String> errors = new ArrayList();
+        List<CheckPaymentDto> updatedPayments = new ArrayList();
+
+        LocalDateTime createdAfterDateTime = null;
+        LocalDateTime createdBeforeDateTime = null;
+        // Define date-time formatter
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+
+        // Parse the parameters if they are provided
+        if( createdAfter != null )
+        {
+            createdAfterDateTime = LocalDateTime.parse(createdAfter, formatter);
+        } else {
+            // startdatetime not provided, check for today
+            createdAfterDateTime = LocalDateTime.now().minusDays(1);
+        }
+
+        if( createdBefore != null )
+        {
+            createdBeforeDateTime = LocalDateTime.parse(createdBefore, formatter);
+        } else {
+            // enddatetime not provided, check for today
+            createdBeforeDateTime = LocalDateTime.now().minusMinutes(5);
+        }
+        log.info("Synchronizing payments between {} and {}", createdAfterDateTime, createdBeforeDateTime);
+
+        try {
+            // get list of payments to synchronize
+            List<CheckPaymentDto> payments = onlinePaymentService.getUnpaidPaymentsToCheck(createdAfterDateTime, createdBeforeDateTime);
+
+            // get paytrail statuses for payments
+            for( CheckPaymentDto payment : payments ) {
+                try {
+                    // get product id from first paymentItem
+                    List<PaymentItem> items = onlinePaymentService.getPaymentItemsForPayment(payment.getPaymentId());
+                    if (items.size() > 0) {
```

Comment:
Make this negation and use continue if size is below 0, so we do not need to nest code.